RecoPixelVertexing/PixelTriplets/plugins/PixelTripletLargeTipGenerator.cc

```diff
@@ -75,7 +75,7 @@ void PixelTripletLargeTipGenerator::fillDescriptions(edm::ParameterSetDescriptio
 
 namespace {
   inline bool intersect(Range& range, const Range& second) {
-    if ((range.min() > second.max()) | (range.max() < second.min()))
+    if ((range.min() > second.max()) || (range.max() < second.min()))
```

Comment:
I think that here the change to a bitwise logic was intentional https://github.com/cms-sw/cmssw/pull/12882/files#diff-bfa7c109eb8f4cb5d2362a1cf482579243cac4b1c08ae48153410f056b83d5f6R63

Please justify that this change is not making the execution slower.
Perhaps a change in syntax that wouldn't change the final compiled code can be done to avoid the warning?

Reply:
godbolt [shows](https://godbolt.org/z/vaEPnr1h3) that this change adds a jump which might slow down the execution. We should look for some other way to silence this warning

Reply:
Thanks, for the moment I can exclude this change from the PR so that we can go ahead with the other changes.

Reply:
I stopped on this file, the other tracking-related code probably still applies, certainly for parts that are executed more than just in configuration or module construction

Reply:
I will do a profiling to compare the executions then